src/sim/SimTrackView.hh

```diff
@@ -80,6 +86,24 @@ CELER_FUNCTION SimTrackView& SimTrackView::operator=(const Initializer_t& other)
     return *this;
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Set the total number of steps.
+ */
+CELER_FUNCTION void SimTrackView::steps(size_type count)
+{
+    states_.state[thread_].steps = count;
+}
```

Comment:
It might be better for our interface to only allow incrementing the step count -- since we never need to set it directly.